src/SmartTransactionsController.ts

```diff
@@ -363,13 +345,47 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       ethQuery = new EthQuery(networkClient.provider);
     }
 
-    this.#updateSmartTransaction(smartTransaction, {
+    this.#createOrUpdateSmartTransaction(smartTransaction, {
       chainId,
       ethQuery,
     });
   }
 
-  async #updateSmartTransaction(
+  #updateSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId = this.config.chainId,
+    }: {
+      chainId: Hex;
+    },
+  ) {
+    const { smartTransactionsState } = this.state;
+    const { smartTransactions } = smartTransactionsState;
+    const currentSmartTransactions = smartTransactions[chainId] ?? [];
+    const currentIndex = currentSmartTransactions?.findIndex(
+      (stx) => stx.uuid === smartTransaction.uuid,
+    );
+    if (currentIndex === -1) {
+      return; // Smart transaction not found, don't update anything.
+    }
+    this.update({
+      smartTransactionsState: {
+        ...smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactionsState.smartTransactions,
+          [chainId]: smartTransactionsState.smartTransactions[chainId].map(
+            (item, index) => {
```

Comment:
Hey, consider naming the item parameter something more specific like existingSmartTransaction to make it clearer.